videoanalyser/app/src/main/java/com/example/videoanalyser/MainActivity.java

```diff
@@ -0,0 +1,132 @@
+package com.example.videoanalyser;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.mlkit.vision.common.InputImage;
+import com.google.mlkit.vision.text.Text;
+import com.google.mlkit.vision.text.TextRecognition;
+import com.google.mlkit.vision.text.TextRecognizer;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+import static android.media.MediaMetadataRetriever.METADATA_KEY_VIDEO_FRAME_COUNT;
+
+public class MainActivity extends AppCompatActivity {
+    private TextView filePath;
+    private Button filePickerBtn;
+    private Button analyzeBtn;
+    private Intent fileIntent;
+    private MediaMetadataRetriever mediaMetadataRetriever;
+    private Uri fileUri;
+    private  Integer totalFrames=0;
```

Comment:
Nit: Please run the formatter over this file. I think this should be "private Integer totalFrames = 0".
Can see some strange whitespace issues further down the file (extra lines before end of functions/anonymous class declarations, odd indenting, etc.).

Reply:
Sure, thanks for pointing!